design-proposals/instancetype.kubevirt.io/custom-resource-sizing.md

```diff
@@ -0,0 +1,149 @@
+
+## Allow for the easy creation of customization common-instancetype resources
+
+### Introduce `virtctl customize`
+
+This new `virtctl` sub-command would allow users to make subtle changes to existing resources within the cluster, outputting a fresh version they can either save locally or apply directly.
+
+Initial support for instance types would be provided and allow the amount of vCPU and RAM supplied by an instance type to be changed. Support for more switches could then follow depending on feedback from users.
+
+The command will also support switching between the cluster and namespaced kinds of instance types and preferences. This should allow a user to take a cluster wide deployed `common-instancetype` such as `u1.medium` and replace it with a namespaced version specific to their workload use case within a namespace they control.
+
+## API Examples
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+[..]
+spec:
+  instancetype:
+    name: cx1.custom
+  template:
+    spec:
+      domain:
+        cpu:
+          guest: 4
+        memory:
+          guest: 4Gi
+---
+apiVersion: instancetype.kubevirt.io/v1beta2
+kind: VirtualMachineClusterInstancetype
+metadata:
+  annotations:
+    instancetype.kubevirt.io/description: |-
+      The CX Series provides exclusive compute resources for compute
+      intensive applications.
+
+      *CX* is the abbreviation of "Compute Exclusive".
+
+      The exclusive resources are given to the compute threads of the
+      VM. In order to ensure this, some additional cores (depending
+      on the number of disks and NICs) will be requested to offload
+      the IO threading from cores dedicated to the workload.
+      In addition, in this series, the NUMA topology of the used
+      cores is provided to the VM.
+    instancetype.kubevirt.io/displayName: Compute Exclusive
+  labels:
+    instancetype.kubevirt.io/class: compute.exclusive
+    instancetype.kubevirt.io/cpu: custom
+    instancetype.kubevirt.io/dedicatedCPUPlacement: "true"
+    instancetype.kubevirt.io/hugepages: "true"
+    instancetype.kubevirt.io/icon-pf: pficon-registry
+    instancetype.kubevirt.io/isolateEmulatorThread: "true"
+    instancetype.kubevirt.io/memory: custom
+    instancetype.kubevirt.io/numa: "true"
+    instancetype.kubevirt.io/vendor: kubevirt.io
+    instancetype.kubevirt.io/version: "1"
+    instancetype.kubevirt.io/size: custom
+    instancetype.kubevirt.io/common-instancetypes-version: v1.2.0
+  name: cx1.custom
+spec:
+  cpu:
+    dedicatedCPUPlacement: true
+    isolateEmulatorThread: true
+    numa:
+      guestMappingPassthrough: {}
+  ioThreadsPolicy: auto
+  memory:
+    hugepages:
+      pageSize: 2Mi
+```
+
+```shell
+$ virtctl customize clusterinstancetype u1.medium --name u1.mysize --kind instancetype --vcpus 4 --memory 4Gi | kubectl apply -f -
+$ virtctl create vm --instancetype virtualmachineinstancetype/u1.mysize [..] | kubectl apply -f
```

Comment:
Can you please add info what the user needs to do, in order to apply these changes to an exsiting VM? (e.g. clear revisionName, etc.)